src/main/HomeScreen.js

```diff
@@ -70,7 +71,8 @@ export default function HomeScreen(props: Props): Node {
         />
       </View>
       <ServerCompatBanner />
-      <ServerPushSetupBanner navigation={navigation} />
+      <ServerNotifsDisabledBanner navigation={navigation} />
+      <ServerNotifsExpiringBanner />
```

Comment:
Can both banners show at once? That seems like it'd be annoying.

Looking at the implementation, I think the answer is that they could only if pushNotificationsEnabledEndTimestamp is non-null while pushNotificationsEnabled is false. That *shouldn't* happen — the server shouldn't do that — but rather than rely on the server in that respect, I think it'd be good to add a little conditional in `pushNotificationsEnabledEndTimestampWarning` so that we ignore the end timestamp if things aren't enabled in the first place. Can be a small prep commit.